internal/cmd/auth.go

```diff
@@ -26,38 +28,46 @@ import (
 func authenticationGRPC(
 	ctx context.Context,
 	logger *zap.Logger,
-	cfg config.AuthenticationConfig,
-	auditLoggingEnabled bool,
-	store storageauth.Store,
-	oplock storageoplock.Service,
+	cfg *config.Config,
+	forceMigrate bool,
 ) (grpcRegisterers, []grpc.UnaryServerInterceptor, func(context.Context) error, error) {
+	if !cfg.Authentication.Enabled() {
+		return grpcRegisterers{}, nil, func(ctx context.Context) error { return nil }, nil
+	}
+
+	db, driver, shutdown, err := getDB(ctx, logger, cfg, forceMigrate)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
 	var (
-		public   = public.NewServer(logger, cfg)
-		register = grpcRegisterers{
+		authCfg    = cfg.Authentication
+		sqlBuilder = fliptsql.BuilderFor(db, driver)
+		store      = authsql.NewStore(driver, sqlBuilder, logger)
```

Comment:
previously we were passing in an already constructed store to this 'constructor', j/w why we are not creating it inside here? same with oplock

Reply:
The main thing is that I wanted to guard the `getDB` call behind the `cfg.Authentication.Enabled()` check. 

Otherwise, that has to be checked down in `grpc.go` before deciding to call this function or not. I could go back to that other way, I am not sure which is cleaner.

What I had before was the root of the nil panic on the auth db, because I wasnt always calling `getDB`. Now I am using this function to get it, but only when some auth component is enabled.